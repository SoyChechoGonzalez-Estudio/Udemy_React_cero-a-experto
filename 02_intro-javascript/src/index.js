import {getHeroesById2} from './bases/08_import-export';

/*const promesa = new Promise((resolve, reject) => {
	setTimeout(() => {
		// Tarea
		// Importar
		const hero = getHeroesById2(3);
		// resolve(hero);
		resolve(hero);
		// reject('No se pudo encontrar el hÃ©roe');
	}, 2000);
});

promesa.then((hero) => {
	console.log(`Heroe ${hero}`);
})
	.catch((err) => {
		console.warn(err);
	});*/

const getHeroByIdAsync = () => {
	return new Promise((resolve, reject) => {
		setTimeout(() => {
			// const hero = getHeroesById2(3);
			// console.log(getHeroesById2(3));
			resolve(getHeroesById2(3));
		}, 2000);
	});
};

getHeroByIdAsync(2)
	.then((hero) => {
		console.log(`Heroe ${hero}`);
	});