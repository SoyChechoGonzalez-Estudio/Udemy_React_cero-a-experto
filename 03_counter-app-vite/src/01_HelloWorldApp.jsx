const HelloWorldApp = () => {
	return (
		<h1>Hola Mundo!!!!!!!!!!!</h1>
	);
};

export {HelloWorldApp};