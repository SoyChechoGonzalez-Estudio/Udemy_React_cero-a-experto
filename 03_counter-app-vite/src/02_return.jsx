const getResult = (a, b) => {
	return a + b;
};

const newMessage = {
	message: 'Hola, QuÃ© tal mi perro',
	title: 'Checho'
};
const FirstApp = () => {
	return (
		<>
			<h1>Hola Mundo</h1>
			<code>{JSON.stringify(newMessage)}</code>
			<h1>{getResult(1, 10)}</h1>
			<p>Soy un subtitulo</p>
		</>
	);
};

export {FirstApp};
