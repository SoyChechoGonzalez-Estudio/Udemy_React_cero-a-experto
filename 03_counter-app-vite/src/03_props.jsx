const FirstApp = ({title, subtitulo}) => {
	return (
		<>
			<h1>{title}</h1>
			<h1>{subtitulo + 1}</h1>
		</>
	);
};

export {FirstApp};