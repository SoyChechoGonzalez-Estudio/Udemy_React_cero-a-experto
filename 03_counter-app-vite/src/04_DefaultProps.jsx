import PropTypes from 'prop-types';

const FirstApp = ({title, subtitulo, name}) => {
	return (
		<>
			<h1>{title}</h1>
			<h1>{subtitulo + 1}</h1>
			<h1>{name}</h1>
		</>
	);
};

FirstApp.propTypes = {
	subtitulo: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
	name: 'Checho Gonz√°lez',
	subtitulo: 'No hay subtitulo',
	title: 'No hay titulo',
};
export {FirstApp};
