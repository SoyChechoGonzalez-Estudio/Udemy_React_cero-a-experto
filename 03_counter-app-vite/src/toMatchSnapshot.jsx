import PropTypes from 'prop-types';

const toMatchSnapshot = ({title, subtitle, name}) => {
	return (
		<>
			<h1>{title}</h1>
			<h1>{subtitle}</h1>
			<h1>{name}</h1>
		</>
	);
};

toMatchSnapshot.propTypes = {
	title: PropTypes.string.isRequired,
	subtitle: PropTypes.string
};
toMatchSnapshot.defaultProps = {
	title: 'Checho Gonz√°lez',
	subtitle: 'No hay title',
	name: 'Checho'
};
export {toMatchSnapshot};