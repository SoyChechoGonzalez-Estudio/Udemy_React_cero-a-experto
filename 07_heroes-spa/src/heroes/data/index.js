export * from './heroes-with-desc.js';