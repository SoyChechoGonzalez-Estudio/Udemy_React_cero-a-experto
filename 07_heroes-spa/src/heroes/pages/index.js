export * from './DC.jsx';
export * from './Marvel.jsx';
export * from './Hero.jsx';
export * from './Search.jsx';
