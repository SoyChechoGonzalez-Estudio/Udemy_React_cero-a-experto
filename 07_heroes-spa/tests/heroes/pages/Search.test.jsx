import {fireEvent, render, screen} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {Search} from '../../../src/heroes';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockedUseNavigate
}));


describe('Pruebas en Search', () => {
	
	test('Debe de mostrarse correctamente con valores por defecto', () => {
		const {container} = render(
			<MemoryRouter>
				<Search />
			</MemoryRouter>
		);
		
		expect(container).toMatchSnapshot();
	});
	
	
	test('Debe de mostrar a Batman y el input con el valor del query string', () => {
		render(
			<MemoryRouter initialEntries={['/search?q=batman']}>
				<Search />
			</MemoryRouter>
		);
		
		const input = screen.getByRole('textbox');
		expect(input.value).toBe('batman');
		
		const img = screen.getByRole('img');
		expect(img.src).toContain('/assets/heroes/dc-batman.jpg');
		
		const alert = screen.getByLabelText('alert-danger');
		expect(alert.style.display).toBe('none');
	});
	
	test('Debe de mostrar un error si no se encuentra el hero (batman123)', () => {
		render(
			<MemoryRouter initialEntries={['/search?q=batman123']}>
				<Search />
			</MemoryRouter>
		);
		
		const alert = screen.getByLabelText('alert-danger');
		expect(alert.style.display).toBe('');
		
	});
	
	test('Debe de llamar el navigate a la pantalla nueva', () => {
		
		const inputValue = 'superman';
		
		render(
			<MemoryRouter initialEntries={['/search']}>
				<Search />
			</MemoryRouter>
		);
		
		const input = screen.getByRole('textbox');
		fireEvent.change(input, {target: {name: 'searchText', value: inputValue}});
		
		const form = screen.getByRole('form');
		fireEvent.submit(form);
		
		expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${inputValue}`);
	});
	
	
});