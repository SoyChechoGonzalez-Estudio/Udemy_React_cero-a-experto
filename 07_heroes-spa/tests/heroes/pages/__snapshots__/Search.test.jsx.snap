// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Search Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    SearchtTest
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching...
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="SearchtTest a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          SearchtTest
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeInLeft"
      >
        SearchtTest a hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeInLeft"
        style="display: none;"
      >
        Sorry!! I donÂ´t have the hero with name 
        <strong />
      </div>
    </div>
  </div>
</div>
`;
